codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -0,0 +1,428 @@
+     * 'UNKNOWN' we must discard all other header values.
+     */
+    private static final HAProxyMessage V2_UNKNOWN_MSG = new HAProxyMessage(
+            HAProxyProtocolVersion.V2, HAProxyCommand.PROXY, HAProxyProxiedProtocol.UNKNOWN, null, null, 0, 0);
+
+    /**
+     * Version 2 proxy protocol message for local requests. Per spec, we should use an unspecified protocol and family
+     * for 'LOCAL' commands. Per spec, when the proxied protocol is 'UNKNOWN' we must discard all other header values.
+     */
+    private static final HAProxyMessage V2_LOCAL_MSG = new HAProxyMessage(
+            HAProxyProtocolVersion.V2, HAProxyCommand.LOCAL, HAProxyProxiedProtocol.UNKNOWN, null, null, 0, 0);
+
+    private final HAProxyProtocolVersion protocolVersion;
+    private final HAProxyCommand command;
+    private final HAProxyProxiedProtocol proxiedProtocol;
+    private final String sourceAddress;
+    private final String destinationAddress;
+    private final int sourcePort;
+    private final int destinationPort;
+
+    /**
+     * Creates a new instance
+     */
+    private HAProxyMessage(
+            HAProxyProtocolVersion protocolVersion, HAProxyCommand command, HAProxyProxiedProtocol proxiedProtocol,
+            String sourceAddress, String destinationAddress, String sourcePort, String destinationPort) {
+        this(
+                protocolVersion, command, proxiedProtocol,
+                sourceAddress, destinationAddress, portStringToInt(sourcePort), portStringToInt(destinationPort));
+    }
+
+    /**
+     * Creates a new instance
+     */
+    private HAProxyMessage(
+            HAProxyProtocolVersion protocolVersion, HAProxyCommand command, HAProxyProxiedProtocol proxiedProtocol,
+            String sourceAddress, String destinationAddress, int sourcePort, int destinationPort) {
+
+        if (proxiedProtocol == null) {
+            throw new NullPointerException("proxiedProtocol");
+        }
+        AddressFamily addrFamily = proxiedProtocol.addressFamily();
+
+        checkAddress(sourceAddress, addrFamily);
+        checkAddress(destinationAddress, addrFamily);
+        checkPort(sourcePort);
+        checkPort(destinationPort);
+
+        this.protocolVersion = protocolVersion;
+        this.command = command;
+        this.proxiedProtocol = proxiedProtocol;
+        this.sourceAddress = sourceAddress;
+        this.destinationAddress = destinationAddress;
+        this.sourcePort = sourcePort;
+        this.destinationPort = destinationPort;
+    }
+
+    /**
+     * Decodes a version 2, binary proxy protocol header.
+     *
+     * @param header                     a version 2 proxy protocol header
+     * @return                           {@link HAProxyMessage} instance
+     * @throws HAProxyProtocolException  if any portion of the header is invalid
+     */
+    static HAProxyMessage decodeHeader(ByteBuf header) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "decodeHeader" is 32 which is greater than 25 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)